js/import/util/metadata.js

```diff
@@ -0,0 +1,162 @@
+      return;
+    }
+
+    if (!mItem) {
+      mappingData.push(newValues);
+    }
+
+    saveImporterSectionsMapping(originalURL, mappingData);
+  }
+};
+
+/**
+ * Get the elements required to create a metadata row in the UI.  Set values if given.  The caller
+ * is to append it to whatever element they deem suitable.
+ * @param mappingData
+ * @param originalURL
+ * @param mapping
+ * @returns {any}
+ */
+const getMetadataRow = (mappingData, originalURL, mapping) => {
+  const metadataRow = createElement('div', { class: 'row', 'data-metadata-id': mapping.id });
+  const nameField = createElement(
+    'sp-textfield',
+    {
+      class: 'metadata-row-name',
+      placeHolder: 'Enter metadata name',
+      value: mapping.name ?? ''
+    }
+  );
+  const valueField = createElement(
+    'sp-textfield',
+    {
+      class: 'metadata-row-value',
+      placeHolder: 'Enter metadata value',
+      value: mapping.value ?? ''
+    }
+  );
+  const urlField = createElement(
+    'sp-textfield',
+    {
+      class: 'metadata-row-condition',
+      placeHolder: 'Enter condition',
+      value: mapping.condition ?? ''
+    }
+  );
+
+  const delButton = getMetadataRowDeleteButton(mappingData, originalURL);
+  metadataRow.append(nameField, valueField, urlField, delButton);
+  nameField.addEventListener('change', (e) => {
+    updateMetadataMapping(e, mappingData, originalURL);
+  });
+  valueField.addEventListener('change', (e) => {
+    updateMetadataMapping(e, mappingData, originalURL);
+  });
+  urlField.addEventListener('change', (e) => {
+    updateMetadataMapping(e, mappingData, originalURL);
+  });
+
+  return metadataRow;
+};
+
+/**
+ * After the mappings are read in, and the detection has been run, set up the metadata
+ * mappings in the customization tab.
+ * @param mappingData
+ * @param originalURL
+ * @param getRowDeleteButton
+ */
+const initializeMetadata = (mappingData, originalURL, getRowDeleteButton) => {
+  getMetadataRowDeleteButton = getRowDeleteButton;
+  const allHidden = METADATA_EDITOR?.querySelectorAll('[class~="hidden"]');
+  allHidden.forEach((he) => {
+    he.classList.remove('hidden');
+  });
+  METADATA_EDITOR?.querySelector('#metadata-click-detection-prompt').classList.add('hidden');
+
+  const mappingRows = METADATA_EDITOR.querySelectorAll('div.row:not(.header)');
+  if (mappingRows.length > 0) {
+    mappingRows.remove();
+  }
+
+  // Display existing metadata values;
+  mappingData
+    .filter((md) => md.mapping === 'metadata')
+    .forEach((metadata) => {
+      const row = getMetadataRow(mappingData, originalURL, metadata);
+      METADATA_EDITOR_SECTIONS.appendChild(row);
+    });
+
+  ADD_METADATA_BUTTON?.addEventListener('click', () => {
+    if (!METADATA_EDITOR) {
+      return;
+    }
+
+    // If there are non-finished rows, do not add another one - focus the first empty field found.
+    const textFields = METADATA_EDITOR.querySelectorAll('.row:not(.header) sp-textfield');
```

Comment:
You can spread the NodeList immediately into an array instead doing an Array.from later.
```suggestion
    const textFields = [...METADATA_EDITOR.querySelectorAll('.row:not(.header) sp-textfield')];
```